IPython/core/interactiveshell.py

```diff
@@ -3309,7 +3312,16 @@ def error_before_exec(value):
             assert self.history_manager is not None
             # Write output to the database. Does nothing unless
             # history output logging is enabled.
-            self.history_manager.store_output(self.execution_count)
+            hm = self.history_manager
+            hm.store_output(self.execution_count)
+            exec_count = self.execution_count
+
+            # Capture MIME outputs and exceptions
+            if result.result:
+                hm.output_mime_bundles[exec_count] = deepcopy(result.result)
+            if result.error_in_exec:
+                hm.exceptions[exec_count] = result.error_in_exec
```

Comment:
Again, I think I would store MIME bundle representation of exceptions instead of exception objects to avoid memory leaks.